jax/api.py

```diff
@@ -1325,9 +1321,12 @@ def jaxpr_maker(*args, **kwargs):
     wrapped = lu.wrap_init(fun)
     jax_args, in_tree = tree_flatten((args, kwargs))
     jaxtree_fun, out_tree = flatten_fun(wrapped, in_tree)
-    pvals = map(pv_like, jax_args)
-    jaxpr, _, _ = pe.trace_to_jaxpr(jaxtree_fun, pvals)
-    return jaxpr
+    in_pvals = map(pv_like, jax_args)
+    jaxpr, out_pvals, consts = pe.trace_to_jaxpr(jaxtree_fun, in_pvals, instantiate=True)
```

Comment:
One theoretical downside to always having `instantiate=True` is that we can't then use `make_jaxpr` to illustrate what gets hoisted out of a function when e.g. we stage it out to XLA. One option might be to add an `instantiate` argument to `make_jaxpr` with a default value of True. Then it has the right default behavior but still lets us make the point about hoisting (which has been useful a time or two in answering JAXers questions IIRC).

I don't have a strong feeling there though, so whichever you think is best!

Reply:
I think that if instantiate=False, the building of the TypedJaxpr does not quite work, because we do not have the types for the constants. 

I actually do not understand how to explain "instantiate", especially at this level of the API.  I could write this, but it would not make sense to the user of this API: 

instantiate: Whether to instantiate constants into abstract values.

Reply:
You're right, I think we need `instantiate=True` to form a TypedJaxpr. Ignore my suggestion!